src/cast.rs

```diff
@@ -81,33 +78,33 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             TyChar => Err(EvalError::InvalidChar(v)),
 
             // No alignment check needed for raw pointers
-            TyRawPtr(_) => Ok(PrimVal::Bytes(v % (1 << self.memory.pointer_size()))),
+            TyRawPtr(_) => Ok(PrimVal::Bytes(v % (1 << (self.memory.pointer_size()*8)))),
```

Comment:
Formatting nit: `x * 8`, not `x*8`.

It also might be nice to have a comment here saying what we're effectively doing (which is truncating to the target usize width, right?).

I was also worried for a second that we'd have `1 << 64` if the target is 64-bit, which would be an overlong shift if the `1` inferred to anything smaller than a 128-bit integer. But I guess since `v: u128`, the RHS of `%` must be `u128` also, and if `1 << n` is `u128`, the `1` must be `u128`, so it won't be an overlong shift? Maybe just throw `1u128` in there explicitly to soothe my brain. :P